src/ShopBundle/objects/db/TShopPaymentHandler/TShopPaymentHandlerPayPal_PayViaLink.class.php

```diff
@@ -156,7 +165,13 @@ public function ProcessIPNRequest($oOrder, $aURLParameter)
         if ($oPaymentParameterList->Length() > 0) {
             while ($oPaymentParameter = $oPaymentParameterList->Next()) {
                 if ('Completed' == $oPaymentParameter->fieldValue) {
-                    $logger->error("PayPal IPN: the txn '{$sTxnId}' has been processed before and was set to completed".print_r($aURLParameter, true).' data: '.$sData);
+                    $logger->info(
+                        "PayPal IPN: the txn '{$sTxnId}' has been processed before and was already set to completed.",
+                        [
+                            'url_parameters' => $aURLParameter,
+                            'data' => $sData,
```

Comment:
Please move sTxnId into the context and add order id and order number to the context

Reply:
ok